main.py

```diff
@@ -1,18 +1,183 @@
+        "full_name": {"type": "string"}
+      },
+      "required": ["name","full_name"]
+    }
+  },
+  "required": ["action","release","repository"]
+}
 
-def pubsub_queue():
-    return os.environ.get('PUBSUB_QUEUE')
+def validate_signature(request):
 
+    # logging.info(json.dumps(dict(request.headers), indent=4))
+    signature_header = request.headers.get('X-Hub-Signature-256')
+    sha_name, github_signature = signature_header.split('=')
+    if sha_name != 'sha256':
+      logging.error('X-Hub-Signature in payload headers was not sha256=****')
+      return False
+
+    # Create our own signature
+    git_webhook_secret = os.environ.get('GIT_MONDO_WEBHOOK_SECRET')
+    local_signature = hmac.new(git_webhook_secret.encode('utf-8'), msg=request.data, digestmod=hashlib.sha256)
+
+    # See if they match
+    return hmac.compare_digest(local_signature.hexdigest(), github_signature)
+
+def validateJson(jsonData):
+    try:
+      validate(instance=jsonData, schema=releasePayloadSchema)
+    except jsonschema.exceptions.ValidationError as err:
+      logging.error("Payload schema error: '{}'".format(err))
+      return False
+    return True
+
+def init_event_configs(release_json):
+    event_cfgs_str = os.environ.get('MONDO_RELEASE_EVENTS')
+    event_cfgs = json.loads(event_cfgs_str)
+    # pprint(event_cfgs)
+
+    for event_cfg in event_cfgs:
+        for assets in release_json['assets']:
+            if (assets['name'].lower() in event_cfg["asset_name"]):
+                y = {
+                    "browser_download_url": assets['browser_download_url'],
+                    "release_name": release_json['name'],
+                    "release_tag": release_json['tag_name'],
+                    "release_date": release_json['published_at']}
+                event_cfg.update(y)
+                # pprint(event_cfg)
+
+    return event_cfgs
+
+def get_event_records(event_cfg):
+    r = requests.get(event_cfg['browser_download_url'])
+    # pprint(r.status_code)
+    r.encoding = 'utf-8'
+    # pprint(r.encoding)
+    content = r.text
+
+    # Reading the downloaded content and turning it into a pandas dataframe
+    df = pd.read_csv(io.StringIO(content), sep='\t').fillna(np.nan).replace([np.nan], [None])
+    # print('\t'+ df.head(5).to_string().replace('\n', '\n\t'))
+
+    # if this is true then there are duplicates for this set (log warning that only the first is kept)
+    if df.duplicated(subset=event_cfg['pk']).agg(max):
+        logging.warning("Duplicate records in "+event_cfg['asset_name'])
+        # print("Duplicate records in "+event_cfg['asset_name'])
+        # keeps only the first unique row for each group.
+        df = df.groupby(event_cfg['pk']).head(1)
+
+    return df.to_dict (orient='records')
+
+# Optional per-message on_delivery handler (triggered by poll() or flush())
+# when a message has been successfully delivered or
+# permanently failed delivery (after retries).
+def acked(err, msg):
```

Comment:
the `msg` parameter here only gets used in a statement that's commented out

Reply:
If the msg parameter is there to meet the `on_delivery` api, there might be a more pythonic way (not sure what that is though... `**kwargs`?) to ignore params that we're not going to use

Reply:
done